internal/pkg/saelastic/utils.go

```diff
@@ -13,17 +15,21 @@ type EventNotification struct {
 	StartsAt    string
 }
 
+var rexForNestedQuote = regexp.MustCompile(`\\\"`)
+var rexForVes = regexp.MustCompile(`"ves":"{(.*)}"`)
+
 //Sanitize ... Issue with json format in events.
 func Sanitize(jsondata string) string {
-	r := strings.NewReplacer("\\\"", "\"",
-		"\"ves\":\"{", "\"ves\":{",
-		"}\"}", "}}",
-		"[", "",
-		"]", "")
-	result := r.Replace(jsondata)
-	result1 := strings.Replace(result, "\\\"", "\"", -1)
-	return result1
+	// if value for key "ves" is string, we convert it to json
+	vesCleaned := jsondata
+	sub := rexForVes.FindStringSubmatch(jsondata)
+	if len(sub) == 2 {
+		nested := rexForNestedQuote.ReplaceAllString(sub[1], `"`)
+		vesCleaned = rexForVes.ReplaceAllString(jsondata, fmt.Sprintf(`"ves":{%s}`, nested))
```

Comment:
This is not fixing the main issue "the summary field with a double-quote",  which is outside ves.  The Ves field is inside the annotation  field.
 annotation:{summary: "" ,ves :{...} } " 
Also, the problem plugin is ovs_events, for some strange reason, it doesn't have ves part. So you don't have a predictable pattern here.
I have a PR for this pending, also you can see the test file  I created for testing and format of JSON coming in.  You could use the fix  I proposed or modify it.

Reply:
Ok, I was not aware of that issue, can definitely add that there.

Reply:
It is the summary field, The PR I pushed address that problem only, I mean it doesn't change anything except removing the double quote from the summary, that way we are not breaking anything that is working.  
In the original code, there was a problem with the summary leaving a double quote if we remove it it will work for all types of the plugin. 

Also, I tried your changes in the playground and all my test failed.  I think you are not considering the possibilities of different plugin types having escaped backslash in the keys. example ``` {\\\"domain\\\":\\\" fault\\\"  ```  

I would suggest trying to enable all plugin and test or try to run the examples with all dirty strings in the Playground to pass your changes, that should be a safe test.
 Playground link https://play.golang.org/p/2012A2prWRF

Also you are not looking at whole string, just ves substring. There are escaped backslash from key fields to be removed. The problem is all string doesn't come with the same number of escape character some with three backslashes and some with one and escaped backslash /// has to be removed from JSON key field.

Reply:
That is because your playground data uses raw strings, eg. they contain extra backslash. This is not true with the current unit test data. I must admit that I did not check if te current unit test data is actually what we get from bus. I will definitely do so.

Reply:
@paramite thats how the data is coming in, with an extra backslash.